webhook.py

```diff
@@ -119,10 +119,27 @@ def send(sender_psid):
             return msbot.constants.RESP_UPDATE_COMPLETE
         return msbot.constants.RESP_INVALID_UNSUBBED
 
+    def recent(sender_psid):
+        if database.user_exists(sender_psid):
+            user = database.get_user_from_id(sender_psid)
+            last_spoiler = database.get_latest_spoiler_id()
+            last_spoil_date = database.get_latest_spoiler_date()
+            spoilers = database.get_all_spoilers_on_date(last_spoil_date)
+            for spoiler in spoilers:
+                send_spoiler_to(user, spoiler)
+            database.update_user(
+                user.user_id,
+                last_updated=last_spoiler,
+                last_spoiled=last_spoiler
```

Comment:
not familiar with the internals, but how come `last_updated` and `last_spoiler` are both assigned to `last_spoiler`? whats the difference between `last_updated` and `last_spoiled?

Reply:
`last_spoiled` is the last actual spoiler image that the user has seen (either through the SEND command or the RECENT command)

`last_updated` is the last spoiler that the user has been notified ABOUT through the update message.

My reasoning for setting them both to last_spoiler was that the RECENT command should make a user indistinguishable to a user that is up-to-date through normal means - that means both last_updated and last_spoiled should be equal to the max spoiler ID.

That said, there shouldn't be a case where last_updated is different from the max spoiler ID barring a race condition between the threads (which probably does eventually need to be dealt with), but I didn't want to enable a situation in which the RECENT command is executed and the last_updated and last_spoiled had somehow diverged.

Thoughts?

Reply:
makes sense, didn't know the underlying structure